src/Graby.php

```diff
@@ -322,7 +338,7 @@ private function doFetchContent(string $url): array
 
         // check site config for single page URL - fetch it if found
         $isSinglePage = false;
-        if ($this->config['singlepage'] && ($singlePageResponse = $this->getSinglePage($html, $effectiveUrl))) {
+        if ($this->config['singlepage'] && null === $prefetchedContent && ($singlePageResponse = $this->getSinglePage($html, $effectiveUrl))) {
```

Comment:
Would not it be reasonable to pass content from web browser but still want Graby to collect multi-page data? Wallabag can always set `singlepage = false` and `multipage = false` to prevent this.

Reply:
The issue is that I don't see how the http `fetch()` call occurring on lines 407 and 757 could work when you provide a prefetched content.

We could consider that the caller (wallabag in my case) should set singlepage=false and multipage=false when providing prefetched content, however there is no way to set these variables as "oneshot operations" afaik.